### ISSUE TITLE
AgContext feature request: Include additonal options under "NOT IN CROP"

### ISSUE THREAD

**Issue by mkutu** [Weed-AI/Weed-AI#578]
Could you include options like "greenhouse", or "potted plants" as a "NOT IN CROP" option in AgContext?

Doing so would help make it more flexible to diverse datasets while increasing specificity for filtering. Even adding tags like "urban" or "dirt road" would broaden the scope of environments while retaining a weed detection focus.

**Comment by jnothman** [comment]
How does this relate to surface cover / soil colour? do you think they're orthogonal?

also @geezacoleman do you think there would be risks to widening the scope beyond agricultural land contexts?

**Comment by mkutu** [comment]
@jnothman, fair question. In short, not sure. But I could see a couple of situations where specifying unique "Crop type", "soil colour" and "soil coverage" beyond a typical agricultural land context might be useful:

1. Regarding your questions about soil coverage and color; one might be planting in a greenhouse using different soil types
    - potting soil
    - pots with mulch
    - or clay soil from the field
![c6295f23-1602-4907-b4d2-73f99faa2876](https://user-images.githubusercontent.com/54868229/147346286-305b9ebb-3a79-4519-b1e9-42587e83b483.JPG)

![IMG_4225](https://user-images.githubusercontent.com/54868229/147346524-deec5f28-d820-407c-b49c-bba25249a2c4.jpg)

Both of these examples above have been used to collect weed images of simple weed scenes that can be easily annotated for detection in the field. The second picture is not the greatest example but you get the idea.

I know we'll be experimenting with different residue types in our potted plants.

2. Regarding your comment to @geezacoleman about extending beyond agricultural land contexts, we have a dataset of Poa annua (debatably one of most invasive turf grass weeds) that might not fit a typical AgContext option because it was grown in a nursery (below). 
![IMG_4224](https://user-images.githubusercontent.com/54868229/147347139-6320e378-8a50-4755-be09-6d9edfc305c1.jpg)

**Comment by jnothman** [comment]
Would you use a mix of soil types in an industry context? or in an experimental/research context?

**Comment by geezacoleman** [comment]
Speaking from my experience, field scenarios are likely to be consistent soil type and surface coverage generally (obviously some variability between localised areas being the nature of agriculture). The scenarios mentioned above are more edge cases for research and while probably more inconvenient, it would be better to upload individually rather than grouped if they are largely different.

On the different types of surface cover front, the ones we have listed were developed with large-scale grain production systems in mind . More flexibility there would be useful if we're involving datasets from more diverse contexts and groups. So adding categories below might address the gaps:
- turfgrass
- pots/potting soil

Potting soil is a slightly more difficult one, because it is technically surface cover: 'None', soil colour: 'whatever suits', but I think it is important to allow people to select for/against these greenhouse scenarios specifically.

> Regarding your comment to @geezacoleman about extending beyond agricultural land contexts, we have a dataset of Poa annua (debatably one of most invasive turf grass weeds) that might not fit a typical AgContext option because it was grown in a nursery (below). We used this dataset to create synthetic data for detection in golf courses

If I were uploading that dataset, it would be surface cover: None, soil colur:'brown' and then describe the collection conditions in your agcontext.

**Comment by jnothman** [comment]
What's the reln between turfgrass and pasture?

**Comment by geezacoleman** [comment]
Pasture could be a very wide and diverse range of species - e.g. DeepWeeds is rangeland 'pasture'. Pastures often have species like clover, ryegrass, medics etc. Likely evidence of grazing or inconsistent growth. Not typically mown at all.

Turfgrass is a very specific set of grass species that aim to have highly consistent growth stage/patterns and are probably mowed. It's likely to be a singular species and won't have any broadleaf plants (unless they are weeds). If this was a platform developed by turfgrass funding you'd probably have > 5 different species listed but perhaps just 'turfgrass' is the specificity we need. @mkutu thoughts?

**Comment by mkutu** [comment]
> If this was a platform developed by turfgrass funding you'd probably have > 5 different species listed but perhaps just 'turfgrass' is the specificity we need. @mkutu thoughts?

Fair point. Would add another major difference would be a lack of soil background for "in-field" examples. Instead you'd have different grasses with different colors (winter dormant, green, yellowing) and different growth stages (mowed, unmowed, sod, think different types of grass lengths on golf course, football fields, etc.). 

> Would you use a mix of soil types in an industry context?

yes, If our goal is to use same or similar models across different areas (eg. parts of the country), which makes sense. I think one of the big issues is the transferability of models across different agricultural landscapes even within the same cropping system (training on images in one place, and inferencing in many others). Changing soil color, types, textures, residues, moisture, cultural practices will be things that we do to capture more variability that we expect to find in the real world. I think especially on the industry side, capturing edge cases that make up the long tail of a data distribution will be important.
